src/general/convex_hull.rs

```diff
@@ -0,0 +1,172 @@
+use std::cmp::Ordering::Equal;
+
+fn sort_by_min_angle(pts: &Vec<(f64, f64)>, min: &(f64, f64)) -> Vec<(f64, f64)> {
+    let mut points: Vec<(f64, f64, (f64, f64))> = pts
+        .iter()
+        .map(|x| {
+            (
+                ((x.1 - min.1) as f64).atan2((x.0 - min.0) as f64),
+                // angle
+                ((x.1 - min.1) as f64).hypot((x.0 - min.0) as f64),
+                // distance (we want the closest to be first)
+                *x,
+            )
+        })
+        .collect();
+    points.sort_by(|a, b| a.partial_cmp(b).unwrap_or(Equal));
+    return points.into_iter().map(|x| x.2).collect();
```

Comment:
This return is not necessary. Clippy calls it a `needless return`. In the future it will be a hard ci error.